cura/Settings/CuraContainerStack.py

```diff
@@ -387,6 +389,18 @@ def getExtruderPositionValueWithDefault(self, key):
             value = int(Application.getInstance().getMachineManager().defaultExtruderPosition)
         return value
 
+    def getProperty(self, key: str, property_name: str, context = None) -> Any:
```

Comment:
Did you look into the `functool.lru_cache` decorator for this as well?
Decorator to wrap a function with a memoizing callable that saves up to the maxsize most recent calls. It can save time when an expensive or I/O bound function is periodically called with the same arguments. Source: https://docs.python.org/3/library/functools.html

How often are the same queries made??

Reply:
Yep. Also discarded it. The LRU cache will only work if for that object the answer will be the same the next time that function is called. We don't have that luxury. (or at least, that's how I understood it)